store/migration/mysql/prod/LATEST.sql

```diff
@@ -127,6 +127,6 @@ CREATE TABLE `reaction` (
   `created_ts` TIMESTAMP NOT NULL DEFAULT CURRENT_TIMESTAMP,
   `creator_id` INT NOT NULL,
   `content_id` VARCHAR(256) NOT NULL,
-  `reaction_type` VARCHAR(256) NOT NULL,
```

Comment:
How about `TINYTEXT` without COLLATE option? I found that it's ok to save emoji into `memo.content` which is `TEXT` type.

Reply:
I didn't set any special charset. But I found:

> MySQL includes character set support that enables you to store data using a variety of character sets and perform comparisons according to a variety of collations. The default MySQL server character set and collation are utf8mb4 and utf8mb4_0900_ai_ci, but you can specify character sets at the server, database, table, column, and string literal levels. To maximize interoperability and future-proofing of your data and applications, we recommend that you use the utf8mb4 character set whenever possible.

https://dev.mysql.com/doc/refman/8.4/en/charset.html#:~:text=The%20default%20MySQL%20server%20character,column%2C%20and%20string%20literal%20levels.

Reply:
In the issue OP mentioned their db charset is `utf8`, not `utf8mb4`. I can set the db default charset to `utf8mb4` so that all users who decide to use a MySQL db align with recommended defaults

Reply:
Adding recommended settings in the docs makes sense for future users.

The emoji reactions feature (and the migrations that go with it) will cause all existing mysql users with this incorrect charset to not be able to upgrade to v0.23 as the migration script will fail to add the default values.

If the migrations are removed then the same subset of users will still not be able to update their settings, use emojis in their memos or add reactions to memos.

For existing users, adding the update in the code seems like the better user experience and should avoid github issues regarding upgrades

Reply:
It shouldn't be needed for new dbs but it looks like there are people who created their dbs back when utf8 (utf8mb3) was the default. 
Issue OP was one of those and today another user had the same issue https://github.com/usememos/memos/issues/4127#issuecomment-2487803176

Are you suggesting to take the db alter out of the latest file and leave it in the migration file?

Reply:
Up to you and @boojack so will go with what you choose.

Out of interest, why do you prefer to not add the db updates?
I can understand that settings `utf8mb4` in `latest` could open users up to being out of date if/when for e.g. `utf8mb5` becomes the new default, but migrations dont have that issue. I feel like I'm missing something.

Reply:
@RoccoSmit It seems to be related to the user's database encoding configuration. We can provide documentation specifying the supported databases and encoding, rather than imposing restrictions in the code. https://stackoverflow.com/a/50264108